src/processing/CommunicateChange.java

```diff
@@ -0,0 +1,90 @@
+package processing;
+
+import java.util.*;
+import processing.IChanges;
+import processing.utility.*;
+
+public class CommunicateChange {
+	
+	// identifier to IChanges handler map
+    // private Map <Identifier, IChanges> identifierTohandler;
+    
+	/*
+     * Computes the set of pixels which will have effect
+     * on the screen and notify the UI using the getChanges
+     * method of IChanges.
+     */
+	
+	// oldPixels -> previous pixels of the object
+	// newPixels -> new pixels of the object
+	public static void provideChanges(
+			Identifier identifier, 
+			ArrayList<Pixel> oldPixels, 
+			ArrayList<Pixel> newPixels) {
+		
+		// stores the set of positions which would have been modified
+		Set <Position> modifiedPos = new HashSet <Position> ();
+		
+		// stores the modified pixels
+		ArrayList <Pixel> modifiedPixels = new ArrayList <Pixel> ();
+		
+		if (oldPixels != null) {
+			// inserts the old position into the set
+			for (Pixel pixel : oldPixels)
+				modifiedPos.add(pixel.position);
+		}
+		
+		if (newPixels != null) {
+			// inserts the new position into the set
+			for (Pixel pixel : newPixels)
+				modifiedPos.add(pixel.position);
+		}
+		
+		// finds the intensity value at each position and push into the array
+		for (Position pos : modifiedPos) {
+			Position curPos = new Position(pos);
+			Intensity intensity;
+			
+			// gets the objectId at pos
+			ObjectId objectId = ClientBoardState
+					.maps
+					.getMostProbableObjectId(
+							new ArrayList <Position> (Arrays.asList(pos))
+							);
```

Comment:
Doubtful regarding multi-line conventions. Please confirm.

Reply:
It's fine